### PR TITLE
upload KeyPoints 2 and 3

### PR DESCRIPTION
**Pull request by federc**
I've just finished topic 0. I've one problem: I can't figure ir out how my default branch is "topic_0" instead of "master", I don't know hot to rename it. Thanks

### CODE REVIEW

File: topic0/src/topic0_KeyPoint2/Building.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	public void setCommon_location(String common_location) {
+		this.common_location = common_location;
+	}
+	public String getPurpose() {
+		return purpose;
+	}
+	public void setPurpose(String purpose) {
+		this.purpose = purpose;
+	}
+
```
Comment (tomiok) [inline topic0/src/topic0_KeyPoint2/Building.java]: add @Override annotation for superclasses's methods.

File: topic0/src/topic0_KeyPoint2/BuildingDirector.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package topic0_KeyPoint2;
+
+public class BuildingDirector {
+	
+	private BuildingBuilder buildingBuilder = null;
```
Comment (tomiok) [inline topic0/src/topic0_KeyPoint2/BuildingDirector.java]: not necesarry to assign to null, if the constructor will be initializing this field.

File: topic0/src/topic0_KeyPoint3/HospitalFactory.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package topic0_KeyPoint3;
+
+public class HospitalFactory {
+	
+	public Hospital getHospital(String type) {
+		if ("about children".equals(type)) {
```
Comment (tomiok) [inline topic0/src/topic0_KeyPoint3/HospitalFactory.java]: Remember we can use equalsIgnoreCase() and don't rely in user or clients lower or upper cases.

Also "about children" could be a constant.

File: topic0/src/topic0_KeyPoint3/InternalMedicinHospital.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package topic0_KeyPoint3;
+
+public class InternalMedicinHospital extends Hospital {
+
+	@Override
+	public String setSpeciality() {
```
Comment (tomiok) [inline topic0/src/topic0_KeyPoint3/InternalMedicinHospital.java]: Bad method naming here, we are getting the speciality instead of setting.

**Comment by federc** [comment]
Corrections made. Also I create "gitignore" file and uploaded it. Have I Already earned the alfajores?

File: topic0/.classpath [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tomiok) [inline topic0/.classpath:1]: Add the .gitignore file to exclude the classpath and compilated files in /target

File: topic0/src/topic0_KeyPoint2/Building.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	public String getPurpose() {
+		return purpose;
+	}
+	
+	public void setPurpose(String purpose) {
+		this.purpose = purpose;
+	}
+	
+	public String toString() {
```
Comment (tomiok) [inline topic0/src/topic0_KeyPoint2/Building.java:24]: Remember to add the @Override annotation for super-classes methods.

**Comment by tomiok** [comment]
@federc I see the gitignore file, but is not ignoring for some reason, after that, consider this PR approved and go ahead and merge it! :+1:

**Comment by tomiok** [comment]
Also remember close mannualy the issues that you worked on.

**Pull request merged**